Sources/_StringProcessing/Engine/Structuralize.swift

```diff
@@ -1,20 +1,38 @@
 internal import _RegexParser
 
-extension CaptureList {
-  @available(SwiftStdlib 5.7, *)
-  func createElements(
-    _ list: MECaptureList
+@available(SwiftStdlib 5.7, *)
+extension Executor {
+  static func createExistentialElements(
+    _ program: MEProgram,
+    matchRange: Range<String.Index>,
+    storedCaptures: [Processor._StoredCapture],
+    wholeMatchValue: Any?
   ) -> [AnyRegexOutput.ElementRepresentation] {
-    assert(list.values.count == captures.count)
-    
+    let capList = program.captureList
+    let capOffsets = program.referencedCaptureOffsets
+
+    // Formal captures include the entire match
+    assert(storedCaptures.count + 1 == capList.captures.count)
+
     var result = [AnyRegexOutput.ElementRepresentation]()
-    
-    for (i, (cap, meStored)) in zip(captures, list.values).enumerated() {
+    result.append(
+      AnyRegexOutput.ElementRepresentation(
+        optionalDepth: 0,
+        content: (matchRange, wholeMatchValue),
+        visibleInTypedOutput: capList.captures[0].visibleInTypedOutput)
+      )
+
+    for (i, (cap, meStored)) in zip(
+      capList.captures.dropFirst(), storedCaptures
+    ).enumerated() {
+      if !cap.visibleInTypedOutput {
+        fatalError("Why does this work?")
+      }
```

Comment:
Leftover?

Reply:
Hmm, I'm still not hitting that fatal error. Did you push the tests that should trigger it?

Reply:
Merged it!